HivePartitioningOptions does not have option to set require partition filter

in bigquery API i see four options mode,uriprefix,requirepartitionfilter & fields.. but in java client i just see two options(first 2).. not sure if this was intentional..

https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#hivepartitioningoptions

requirePartitionFilter (boolean)

enabling requirepartition filter out side of hivepartitionoptions throws out error ---> Caused by: com.google.api.client.googleapis.json.GoogleJsonResponseException: 400 Bad Request
{
  "code" : 400,
  "errors" : [ {
    "domain" : "global",
    "message" : "require_partition_filter has been set for a hive-partitioned table without using the HivePartitioningOptions. To configure external, hive-partitioned tables -- including setting require_partition_filter -- please use the HivePartitioningOptions. Setting require_partition_filter on the top-level table definition, or via the TimePartitioning field, does not configure hive-partitioned tables.",
    "reason" : "invalid"
  } ],
  "message" : "require_partition_filter has been set for a hive-partitioned table without using the HivePartitioningOptions. To configure external, hive-partitioned tables -- including setting require_partition_filter -- please use the HivePartitioningOptions. Setting require_partition_filter on the top-level table definition, or via the TimePartitioning field, does not configure hive-partitioned tables.",
  "status" : "INVALID_ARGUMENT"
}

--- a/google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/HivePartitioningOptions.java
+++ b/google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/HivePartitioningOptions.java
@@ -23,17 +23,20 @@ import java.util.Objects;
 public final class HivePartitioningOptions {
 
   private final String mode;
+  private final Boolean requirePartitionFilter;
   private final String sourceUriPrefix;
 
   public static final class Builder {
 
     private String mode;
+    private Boolean requirePartitionFilter;
     private String sourceUriPrefix;
 
     private Builder() {}
 
     private Builder(HivePartitioningOptions options) {
       this.mode = options.mode;
+      this.requirePartitionFilter = options.requirePartitionFilter;
       this.sourceUriPrefix = options.sourceUriPrefix;
     }
 
@@ -47,6 +50,17 @@ public final class HivePartitioningOptions {
      */
     public Builder setMode(String mode) {
       this.mode = mode;
+      return this;
+    }
+
+    /**
+     * [Optional] If set to true, queries over this table require a partition filter that can be
+     * used for partition elimination to be specified. Note that this field should only be true when
+     * creating a permanent external table or querying a temporary external table. Hive-partitioned
+     * loads with requirePartitionFilter explicitly set to true will fail.
+     */
+    public Builder setRequirePartitionFilter(Boolean requirePartitionFilter) {
+      this.requirePartitionFilter = requirePartitionFilter;
       return this;
     }
 
@@ -72,12 +86,21 @@ public final class HivePartitioningOptions {
 
   private HivePartitioningOptions(Builder builder) {
     this.mode = builder.mode;
+    this.requirePartitionFilter = builder.requirePartitionFilter;
     this.sourceUriPrefix = builder.sourceUriPrefix;
   }
 
   /* Returns the mode of hive partitioning */
   public String getMode() {
     return mode;
+  }
+
+  /**
+   * Returns true if a partition filter (that can be used for partition elimination) is required for
+   * queries over this table.
+   */
+  public Boolean getRequirePartitionFilter() {
+    return requirePartitionFilter;
   }
 
   /* Returns the sourceUriPrefix of hive partitioning */
@@ -99,6 +122,7 @@ public final class HivePartitioningOptions {
   public String toString() {
     return MoreObjects.toStringHelper(this)
         .add("mode", mode)
+        .add("requirePartitionFilter", requirePartitionFilter)
         .add("sourceUriPrefix", sourceUriPrefix)
         .toString();
   }
@@ -109,6 +133,8 @@ public final class HivePartitioningOptions {
         || obj != null
             && obj.getClass().equals(HivePartitioningOptions.class)
             && Objects.equals(mode, ((HivePartitioningOptions) obj).getMode())
+            && Objects.equals(
+                requirePartitionFilter, ((HivePartitioningOptions) obj).getRequirePartitionFilter())
             && Objects.equals(
                 sourceUriPrefix, ((HivePartitioningOptions) obj).getSourceUriPrefix());
   }
@@ -122,6 +148,7 @@ public final class HivePartitioningOptions {
     com.google.api.services.bigquery.model.HivePartitioningOptions options =
         new com.google.api.services.bigquery.model.HivePartitioningOptions();
     options.setMode(mode);
+    options.setRequirePartitionFilter(requirePartitionFilter);
     options.setSourceUriPrefix(sourceUriPrefix);
     return options;
   }
@@ -131,6 +158,9 @@ public final class HivePartitioningOptions {
     Builder builder = newBuilder();
     if (options.getMode() != null) {
       builder.setMode(options.getMode());
+    }
+    if (options.getRequirePartitionFilter() != null) {
+      builder.setRequirePartitionFilter(options.getRequirePartitionFilter());
     }
     if (options.getSourceUriPrefix() != null) {
       builder.setSourceUriPrefix(options.getSourceUriPrefix());

--- a/google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/HivePartitioningOptionsTest.java
+++ b/google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/HivePartitioningOptionsTest.java
@@ -24,9 +24,11 @@ public class HivePartitioningOptionsTest {
 
   private static final String MODE = "STRING";
   private static final String SOURCE_URI_PREFIX = "gs://bucket/path_to_table";
+  private static final Boolean REQUIRE_PARTITION_FILTER = true;
   private static final HivePartitioningOptions HIVE_PARTITIONING_OPTIONS =
       HivePartitioningOptions.newBuilder()
           .setMode(MODE)
+          .setRequirePartitionFilter(REQUIRE_PARTITION_FILTER)
           .setSourceUriPrefix(SOURCE_URI_PREFIX)
           .build();
 
@@ -49,6 +51,8 @@ public class HivePartitioningOptionsTest {
   @Test
   public void testBuilder() {
     assertThat(HIVE_PARTITIONING_OPTIONS.getMode()).isEqualTo(MODE);
+    assertThat(HIVE_PARTITIONING_OPTIONS.getRequirePartitionFilter())
+        .isEqualTo(REQUIRE_PARTITION_FILTER);
     assertThat(HIVE_PARTITIONING_OPTIONS.getSourceUriPrefix()).isEqualTo(SOURCE_URI_PREFIX);
   }
 
@@ -62,6 +66,7 @@ public class HivePartitioningOptionsTest {
   private void compareHivePartitioningOptions(
       HivePartitioningOptions expected, HivePartitioningOptions value) {
     assertThat(value.getMode()).isEqualTo(expected.getMode());
+    assertThat(value.getRequirePartitionFilter()).isEqualTo(expected.getRequirePartitionFilter());
     assertThat(value.getSourceUriPrefix()).isEqualTo(expected.getSourceUriPrefix());
     assertThat(value.toString()).isEqualTo(expected.toString());
     assertThat(value.hashCode()).isEqualTo(expected.hashCode());

--- a/pom.xml
+++ b/pom.xml
@@ -63,7 +63,7 @@
     <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
     <github.global.server>github</github.global.server>
     <site.installationModule>google-cloud-bigquery-parent</site.installationModule>
-    <google-api-services-bigquery.version>v2-rev20200916-1.30.10
+    <google-api-services-bigquery.version>v2-rev20200925-1.30.10
     </google-api-services-bigquery.version>
   </properties>
 

